### ISSUE TITLE
Veto's can come in a little hot

### ISSUE THREAD

**Issue by miped** [splewis/get5#122]
A lot of CS players have what seems to amount to a nervous tick with constantly switching weapons (i.e. pressing  1 through 5 constantly). When running MWL, we were very clear that veto's would be happening as soon as the last player readied up, and that we would not be redoing veto's because people weren't paying attention and vetoed maps by mistake.

Quite a number of people hit that situation, and we've gotten feedback both at the event, and in the post-event survey that the veto does come in a little bit hot. While we could just take a hardline approach on this, i think there are some easy ways to make it a little more foolproof.

They are not mutually exclusive, so i think we should add both of them.

Proposal 1
----------
Do a 5 second countdown and a restart of the match (still in warmup mode), but freeze all the players while the veto is happening. Haven't looked into how easy this would be to implement, but there are several examples in the code about restarting the game, so should be fairly straight forward.

Basically, there would be messages to all players, and the game would restart which would go something like
```
[Get5] Veto coming up in 5...
[Get5] Veto coming up in 4...
[Get5] Veto coming up in 3...
[Get5] Veto coming up in 2...
[Get5] Veto coming up in 1...
<game restarts>
<match pauses during freezetime>
[Get5] Veto started. Game will unpause when veto is complete.
```
And the game will be unpaused if veto is aborted (on a captain drop). If veto is finished successfully, then the map change will happen right after, and there's no reason to unpause the game.

Proposal 2
-----------
Once one of the veto menus is presented, record a timestamp. When the captain does a selection, figure out how fast that pick was made. If they selected an option in less than 3 seconds, present another menu that goes like this:
```
Are you sure you want to ban Cache:
  8. Yes, ban Cache
  9. No, go back
```

The important thing here is that the options are on buttons which most players don't use very often. And adding the timer means that if they actually thought about it, they won't be presented with this menu.

**Comment by Technoblazed** [comment]
The first half of `Option 1` would be the better option, but not the restarting of the game and pausing section of it, as warmup should still continue during the veto stages.

**Comment by miped** [comment]
I don't agree, for the following reasons:
 - Veto is important, and everyone should be paying attention and communicating. Not running around warming up.
 - There will be another round of warmup once the veto is done. Players should be able to chill out for like 2 min while vetoing is happening.
 - Leaving the game running means that even the person doing the veto for the team would be tempted to keep warming up while the other team is vetoing. Thus, he'll be pressing his number keys to switch weapons, except the other team just finished their pick, and oops, now he's selected a map they didn't want to play, which is exactly what we're trying to avoid.

**Comment by Technoblazed** [comment]
https://github.com/splewis/get5/pull/123/commits/e15da3fa82ba8870a661fe6f8ec6a8b7c87b6249 Something like that then? With just the restart added in

**Comment by miped** [comment]
Actually, thinking about it, i think as long as we add proposal 2 as well, the restart probably isn't needed.

**Comment by miped** [comment]
Would like @splewis input though :-)

**Comment by splewis** [comment]
I agree this is an issue. I like proposal 1.

I think an alternative to the "are you sure?" menu would be displaying another menu before each option where:
- it's the same menu as the "real" menu 
- every option is set to the disabled
- lasts for a second, then the real menu shows up with selectable options

But that's kind of awkward too. I think my preference would be proposal 1, and maybe... making sure the player's don't have money so they won't even have weapons during the veto, and won't even have the urge to spam 1/2 to switch guns .

**Comment by Technoblazed** [comment]
Or, even just block 1/2 from being pushed in the menu?

**Comment by splewis** [comment]
I'd like to have at least 7 maps fit in the menu, and I think removing 1+2 would mean there isn't room for all the maps.

**Comment by Technoblazed** [comment]
Radio menus are 10 per page,
so:

```
1 - Blocked
2 - Blocked
3 - Map 1
4 - Map 2
5 - Map 3
6 - Map 4
7 - Map 5
8 - Map 6
9 - Map 7
10 - Next Page

**Comment by miped** [comment]
And people don't just spam 1 and 2 unfortunately... Anything 5 and below is in danger.

**Comment by Technoblazed** [comment]
Well, the captains are the only people that get the veto panel, so another method would just be to strip weapons from the client with a warning and then provide the veto menu, idk.

**Comment by miped** [comment]
I found a way to add confirmation dialogs to the veto that i think is pretty unobtrusive, while being configurable enough so that it can be turned off (default), activated in some cases (like what we're looking for at Mad City), or always activated (if that's your thing). Options to confirm or deny are on buttons 8 and 9 respectively, so they should be out of the danger zone.

Here's how it looks: http://d.pr/i/aOpw (low time setting which is the reason it doesn't prompt for confirmation the second time).

I think that this pull request, combined with @Technoblazed's will make the veto process completely fool proof, just by setting the right cvars.

**Comment by Technoblazed** [comment]
I like it, I like it a lot.

**Comment by miped** [comment]
Fixed by #122 and #126
